File: mvvmsmart/src/main/java/com/wzq/mvvmsmart/http/interceptor/ProgressInterceptor.java
package com.wzq.mvvmsmart.http.interceptor;

import com.wzq.mvvmsmart.http.download.ProgressResponseBody;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Response;



public class ProgressInterceptor implements Interceptor {

    @Override
    public Response intercept(Chain chain) throws IOException {
        Response originalResponse = chain.proceed(chain.request());
        return originalResponse.newBuilder()
                .body(new ProgressResponseBody(originalResponse.body()))
                .build();
    }
}

File: mvvmsmart/src/main/java/com/wzq/mvvmsmart/http/interceptor/ProgressInterceptor.kt
package com.wzq.mvvmsmart.http.interceptor

import com.wzq.mvvmsmart.http.download.ProgressResponseBody
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class ProgressInterceptor : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val originalResponse = chain.proceed(chain.request())
        return originalResponse.newBuilder()
                .body(ProgressResponseBody(originalResponse.body()))
                .build()
    }
}